### PR TITLE
Improve exists() via commonFileExtensions

### PR DESCRIPTION
**Pull request by ruhan1**
If path ends with specific extensions, we know it is file. We can improve exists() performance to mitigate the timeout problem.

### CODE REVIEW

**Review by sswguo — COMMENTED** [review]
LGTM.

File: storage/src/main/java/org/commonjava/storage/pathmapped/core/PathMappedFileManager.java [hunk 1]
```diff
@@ -242,6 +246,12 @@ public boolean exists( String fileSystem, String path )
         {
             return false;
         }
+
+        if ( commonFileExtensions != null && path.matches( commonFileExtensions ) )
+        {
+            return pathDB.existsFile( fileSystem, path );
+        }
+
```
Comment (ligangty) [inline storage/src/main/java/org/commonjava/storage/pathmapped/core/PathMappedFileManager.java]: Seems this lacks of physical file exists checking.

Comment (ruhan1) [reply storage/src/main/java/org/commonjava/storage/pathmapped/core/PathMappedFileManager.java]: Actually I hesitate to add the physical file check. The pathDB should be the only source-of-truth. If pathDB entry exists but physical file missing, that indicate a bug and IOException should throw. How about running this code to see how it goes?

Comment (ligangty) [reply storage/src/main/java/org/commonjava/storage/pathmapped/core/PathMappedFileManager.java]: I remember we have a bug before which is caused this type of problem, so I'm not sure here. But if you prefer to test this, I'd prefer to comment the physical file checking code in this part of code and add some comments to indicate the reason.

Comment (ruhan1) [reply storage/src/main/java/org/commonjava/storage/pathmapped/core/PathMappedFileManager.java]: Yeah it is good point. I added some comments. If that problem happen again, I guess we have to find the root cause.

Comment (ligangty) [reply storage/src/main/java/org/commonjava/storage/pathmapped/core/PathMappedFileManager.java]: So maybe we should consolidate the exists checking here, that means whatever if it is a common case or not, the physical file check should be ignored. If you think this is an accepted case, I'd recommend we can remove or comment out the physical file checking code here with your comment above.

Comment (ruhan1) [reply storage/src/main/java/org/commonjava/storage/pathmapped/core/PathMappedFileManager.java]: Ok. let's do it. The exists performance is pushing. I updated pr.

**Comment by ruhan1** [comment]
It seems we also need pathDB.getStorageFile in exist() to check expiration. Although this is not used atm, I don't want to lose the feature.

**Pull request merged**